src/java.base/share/classes/java/lang/foreign/ValueLayout.java

```diff
@@ -649,6 +649,126 @@ public boolean isUnbounded() {
         }
     }
 
+    /**
+     * This class contains a number of unaligned value layouts.
+     * <p>
+     * @apiNote Care should be taken when using unaligned value layouts as they may induce
+     *          performance and portability issues.
+     */
+    public static final class Unaligned {
+        // These less frequently used constants are lazily initialized and not directly visible in ValueLayout
+        private Unaligned() { }
+
+        /**
+         * An unaligned value layout constant whose size is the same as that of a machine address ({@code size_t}),
+         * and byte order set to {@link ByteOrder#nativeOrder()}.
+         * Equivalent to the following code:
+         * {@snippet lang=java :
+         * MemoryLayout.valueLayout(MemorySegment.class, ByteOrder.nativeOrder());
+         * }
+         * @apiNote Care should be taken when using unaligned value layouts as they may induce
+         *          performance and portability issues.
+         */
+        public static final OfAddress UNALIGNED_ADDRESS = new OfAddress(ByteOrder.nativeOrder());
```

Comment:
This seems to create an address layout with the alignment set to the size of the layout? For unaligned layouts we want the alignment set to 1 byte/8 bits I think.

Reply:
I'd like not to have a separate nested class if possible. That will show up in javadoc and make those constants less discoverable.

Also, I think it would be better to use UNALIGNED as a suffix - e.g. JAVA_INT_UNALIGNED.

Reply:
> This seems to create an address layout with the alignment set to the size of the layout? For unaligned layouts we want the alignment set to 1 byte/8 bits I think.

You are right. So, parts like `new OfInt(ByteOrder.nativeOrder()).withBitAlignment(32)` could be simplified to `new OfInt(ByteOrder.nativeOrder())`.